sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java

```diff
@@ -1135,8 +1138,22 @@ public DataLakeFileOpenInputStreamResult openInputStream(DataLakeFileInputStream
      * @throws DataLakeStorageException If a storage service error occurred.
      */
     public DataLakeFileOpenInputStreamResult openInputStream(DataLakeFileInputStreamOptions options, Context context) {
+        Context newContext;
+        options = options == null ? new DataLakeFileInputStreamOptions() : options;
```

Comment:
Do we need to new up `DataLakeFileInputStreamOptions` here? If it's null shouldn't `options.isUpn()` be null?

Could we just make the if check below

`if (options != null && options.isUpn() != null) {`